zebra-chain/src/serialization.rs

```diff
@@ -55,6 +55,31 @@ pub trait ZcashDeserialize: Sized {
     fn zcash_deserialize<R: io::Read>(reader: R) -> Result<Self, SerializationError>;
 }
 
+impl<T: ZcashSerialize> ZcashSerialize for Vec<T> {
+    fn zcash_serialize<W: io::Write>(&self, mut writer: W) -> Result<(), SerializationError> {
+        writer.write_compactsize(self.len() as u64)?;
+        for x in self {
```

Comment:
Yep, but this is only possible because the impl is defined in the same crate as the trait; otherwise, this would be disallowed by the orphan rules (so that there can never be overlapping trait impls).